src/c++/perf_analyzer/report_writer.cc

```diff
@@ -388,4 +402,56 @@ ReportWriter::WriteGpuMetrics(std::ostream& ofs, const Metrics& metric)
   }
 }
 
+void
+ReportWriter::WriteLLMMetrics(std::ostream& ofs)
+{
+  auto [avg_first_token_latency, avg_t2t_latency] = CalculateLLMMetrics();
+
+  if (avg_first_token_latency.has_value()) {
+    ofs << "," << avg_first_token_latency.value();
+  } else {
+    ofs << ",n/a";
+  }
+  if (avg_t2t_latency.has_value()) {
+    ofs << "," << avg_t2t_latency.value();
+  } else {
+    ofs << ",n/a";
+  }
+}
+
+std::tuple<std::optional<double>, std::optional<double>>
+ReportWriter::CalculateLLMMetrics()
+{
+  if (collector_->IsEmpty()) {
+    throw PerfAnalyzerException(
+        "Attempted to write LLM metrics when profile data is empty.",
+        GENERIC_ERROR);
+  }
+
+  const std::vector<Experiment>& experiments{collector_->GetData()};
+  std::vector<double> first_token_latencies;
+  std::vector<double> t2t_latencies;
+
+  for (const auto& exp : experiments) {
```

Comment:
How do experiments work? My understanding is that running with a request rate of 2 and then running with a request rate of 4 is two separate experiments. Do we really want to combine the stats of all experiments into a single number? Or am I misunderstanding the code? I haven't really run PA with multiple experiments, but I would expect each experiment to have its own results and metrics.

Reply:
Great catch 👍 You're right. This was under my radar because I was always just running a single experiment. Changed the code to handle multiple experiments

Reply:
Great.
- Is there any new per-experiment testing?
- Can you show what the output looks like for a single experiment and for a multiple-experiment run?

Reply:
Added a testcase for `GenerateReport()` which includes single multiple experiments cases.